api/src/db/models/opportunity_models.py

```diff
@@ -49,72 +49,65 @@ class Opportunity(Base, TimestampMixin):
-        back_populates="opportunity", uselist=True, cascade="all, delete-orphan"
+
+    current_opportunity_summary: Mapped["CurrentOpportunitySummary | None"] = relationship(
+        back_populates="opportunity", single_parent=True, cascade="all, delete-orphan"
     )
 
     @property
-    def funding_instruments(self) -> list[FundingInstrument]:
-        # Helper method for serialization of the API response
-        return [f.funding_instrument for f in self.link_funding_instruments]
+    def summary(self) -> "OpportunitySummary | None":
+        """
+        Utility getter method for converting an Opportunity in our endpoints
 
-    @property
-    def funding_categories(self) -> list[FundingCategory]:
-        # Helper method for serialization of the API response
-        return [f.funding_category for f in self.link_funding_categories]
+        This handles mapping the current opportunity summary to the "summary" object
+         in our API responses - handling nullablity as well.
+        """
+        if self.current_opportunity_summary is None:
+            return None
 
-    @property
-    def applicant_types(self) -> list[ApplicantType]:
-        # Helper method for serialization of the API response
-        return [a.applicant_type for a in self.link_applicant_types]
+        return self.current_opportunity_summary.opportunity_summary
 
 
 class OpportunitySummary(Base, TimestampMixin):
     __tablename__ = "opportunity_summary"
 
-    opportunity_id: Mapped[int] = mapped_column(
-        ForeignKey(Opportunity.opportunity_id), primary_key=True
-    )
-    opportunity: Mapped[Opportunity] = relationship(Opportunity)
+    opportunity_summary_id: Mapped[int] = mapped_column(primary_key=True)
 
-    opportunity_status: Mapped[OpportunityStatus | None] = mapped_column(
-        "opportunity_status_id",
-        LookupColumn(LkOpportunityStatus),
-        ForeignKey(LkOpportunityStatus.opportunity_status_id),
-    )
+    opportunity_id: Mapped[int] = mapped_column(ForeignKey(Opportunity.opportunity_id))
+    opportunity: Mapped[Opportunity] = relationship(Opportunity)
 
     summary_description: Mapped[str | None]
+
     is_cost_sharing: Mapped[bool | None]
+    is_forecast: Mapped[bool | None]
 
+    post_date: Mapped[date | None]
     close_date: Mapped[date | None]
     close_date_description: Mapped[str | None]
-
-    post_date: Mapped[date | None]
     archive_date: Mapped[date | None]
     unarchive_date: Mapped[date | None]
 
+    # The award amounts can be for several billion requiring us to use BigInteger
     expected_number_of_awards: Mapped[int | None]
-    estimated_total_program_funding: Mapped[int | None]
-    award_floor: Mapped[int | None]
-    award_ceiling: Mapped[int | None]
+    estimated_total_program_funding: Mapped[int | None] = mapped_column(BigInteger)
+    award_floor: Mapped[int | None] = mapped_column(BigInteger)
+    award_ceiling: Mapped[int | None] = mapped_column(BigInteger)
```

Comment:
Did some digging through the data and found several opportunities over the ~4 billion limit of an ordinary integer in Postgres.

For example, one for about $14 billion:
https://grants.gov/search-results-detail/349234

I did test that writing to the DB with very large values works uneventfully (note Python itself doesn't have a practical limit on integers)